test/integration/server-selection/readpreference.test.ts

```diff
@@ -463,4 +457,214 @@ describe('ReadPreference', function () {
       await client.close();
     }
   });
+
+  context('when connecting to a secondary in a replica set with a direct connection', () => {
+    context('when readPreference is primary', () => {
+      it('should attach a read preference of primaryPreferred to the read command for replicaset', {
+        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
+        test: async function () {
+          if (this.configuration.topologyType !== 'ReplicaSetWithPrimary') {
+            return this.skip();
```

Comment:
Should we set a skipReason here? `this.skipReason = '...'` will print out why this test didn't run